conn.go

```diff
@@ -147,7 +150,8 @@ func (d *Daemon) handleConn(c net.Conn) {
 			}
 
 		case pb.Request_PERSISTENT_CONN_UPGRADE:
-			d.handlePersistentConn(r, w)
+			var clearUp sync.Once
```

Comment:
Yes, I think you are right. Here is another problem, which I tried to fix this way (and failed).
Suppose we have balanced handler with several handlers (writers). After disconnection of the first one we will destroy handler for entire protocol.

I think in defer we need to delete only one writer, check if it is the last writer for the protocol and only then remove handler and entire registry.
Does this make sense?

Reply:
I think the current behavior is the best because:

- Currently, `utils.RoundRobin` can't remove entries, so we'd get a handler that fails with probability 1/N. This is worse than a handler that always fails, since it's easier to spot and debug issues in the latter case.

- If we implement removing entries in `utils.RoundRobin`, a handler's death may go unnoticed and we'd get a degraded performance without notice. This would also be harder to spot and debug.